token-group/interface/src/instruction.rs

```diff
@@ -0,0 +1,308 @@
+//! Instruction types
+
+use {
+    bytemuck::{Pod, Zeroable},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    spl_discriminator::{ArrayDiscriminator, SplDiscriminate},
+    spl_pod::{
+        bytemuck::{pod_bytes_of, pod_from_bytes},
+        optional_keys::OptionalNonZeroPubkey,
+    },
+};
+
+/// Instruction data for initializing a new `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:initialize_group")]
+pub struct InitializeGroup {
+    /// Update authority for the group
+    pub update_authority: OptionalNonZeroPubkey,
+    /// The maximum number of group members
+    pub max_size: u32,
```

Comment:
You'll likely need to use `PodU32` for the instructions and structs to avoid padding / alignment issues